clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/configuration/condition/ConditionalOnCamundaClientEnabled.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.spring.client.configuration.condition;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import org.springframework.context.annotation.Conditional;
+
+@Target({TYPE, METHOD})
+@Retention(RUNTIME)
+@Documented
+@Conditional(CamundaClientEnabledCondition.class)
```

Comment:
Since the condition is only on a property, we can use Spring's `@ConditionalOnProperty`. No need to have `CamundaClientEnabledCondition`
```suggestion
@ConditionalOnProperty(name = "camunda.client.enabled", havingValue = "true", matchIfMissing = true)
```